src/Daemon.php

```diff
@@ -67,13 +70,12 @@ public function setListener(callable $listener)
      */
     public function run()
     {
-        $env = $this->context->env;
-        $tube = $env->get('BEANSTALKD_TUBE') ?: 'default';
-        $class = $env->get('BEANSTALKD_CONSUMER');
-        if (!$class) {
+        if (!$this->env->hasValue(self::CONSUMER_KEY)) {
             $this->stdio->outln('<<red>>BEANSTALKD_CONSUMER environmental variable is not set<<reset>>');
             return Status::USAGE;
         }
+
+        $class = $this->env->getValue(self::CONSUMER_KEY);
         if (!class_exists($class)) {
             $this->stdio->outln(sprintf('<<red>>BEANSTALKD_CONSUMER does not reference a locatable class: %s<<reset>>', $class));
```

Comment:
Right, but having the Env leak into domain logic is not ideal. Since these values are required, it makes sense to have them checked/defined before it ever gets to this point. Being able to do the following would be more consistent, imo:

```php
$consumer = $this->config->consumer();
```

Then you have a fully resolved instance without having to inject `Env` or `ResolverInterface`.